common/pkg/libartifact/store/store_test.go

```diff
@@ -0,0 +1,838 @@
+	err = os.WriteFile(filePath, content, 0o644)
+	require.NoError(t, err)
+
+	return libartTypes.ArtifactBlob{
+		BlobFilePath: filePath,
+		FileName:     fileName,
+	}
+}
+
+// helperAddArtifact is a test helper that adds an artifact to the store.
+// It creates temporary files with random content and adds them as blobs.
+// fileNames maps filename to size in bytes of random content to generate.
+// If options is nil, uses a default with ArtifactMIMEType set to "application/vnd.test+type".
+func helperAddArtifact(t *testing.T, as *ArtifactStore, refName string, fileNames map[string]int, options *libartTypes.AddOptions) *digest.Digest {
+	t.Helper()
+	ctx := context.Background()
+
+	// If options is nil, create default options
+	if options == nil {
+		options = &libartTypes.AddOptions{
+			ArtifactMIMEType: "application/vnd.test+type",
+		}
+	}
+
+	// if no specific files were passed, create a random file of 2k
+	if fileNames == nil {
+		filename, err := randomAlphanumeric(5)
+		require.NoError(t, err)
+		fileNames = map[string]int{
+			filename: 2,
+		}
+	}
+
+	// Create artifact reference
+	ref, err := NewArtifactReference(refName)
+	require.NoError(t, err)
+
+	// Create artifact blobs
+	blobs := make([]libartTypes.ArtifactBlob, 0, len(fileNames))
+	for fileName, size := range fileNames {
+		blobs = append(blobs, createTestBlob(t, fileName, size))
+	}
+
+	// Add artifact
+	artifactDigest, err := as.Add(ctx, ref, blobs, options)
+	require.NoError(t, err)
+	require.NotNil(t, artifactDigest)
+
+	return artifactDigest
+}
+
+func TestNewArtifactStore(t *testing.T) {
+	// Test with valid absolute path
+	storePath := filepath.Join(t.TempDir(), "store")
+	sc := &types.SystemContext{}
+
+	as, err := NewArtifactStore(storePath, sc)
+	assert.NoError(t, err)
+	assert.NotNil(t, as)
+	assert.Equal(t, storePath, as.storePath)
+
+	// Verify the index file was created
+	indexPath := filepath.Join(storePath, "index.json")
+	_, err = os.Stat(indexPath)
+	assert.NoError(t, err)
+
+	// Test with empty path
+	_, err = NewArtifactStore("", sc)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "store path cannot be empty")
+
+	// Test with relative path
+	_, err = NewArtifactStore("relative/path", sc)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "must be absolute")
+}
+
+func TestArtifactStore_Add(t *testing.T) {
+	as, ctx := setupTestStore(t)
+
+	// Add artifact using helper with nil options (uses default)
+	fileNames := map[string]int{
+		"testfile.txt": 1024,
+	}
+
+	refName := "quay.io/test/artifact:v1"
+	artifactDigest := helperAddArtifact(t, as, refName, fileNames, nil)
+	assert.NotEmpty(t, artifactDigest.String())
+
+	// Verify artifact was added to the store
+	artifacts, err := as.List(ctx)
+	require.NoError(t, err)
+	assert.Len(t, artifacts, 1)
+
+	// Verify artifact properties
+	artifact := artifacts[0]
+	assert.Equal(t, refName, artifact.Name)
+	assert.Equal(t, "application/vnd.test+type", artifact.Manifest.ArtifactType)
```

Comment:
(This is sort of a remote dependency on `helperAddArtifact`’s default, but, as a test of the storage persisting data, it works well enough.)